### PR TITLE
chore(app): delete old code to lighten webpack bundle

### PR DESCRIPTION
**Pull request by lmiller1990**
- Closes https://cypress-io.atlassian.net/browse/UNIFY-1047

### User facing changelog

N/A

### Additional details

- move more code from `runner-shared` to `app` to reduce reliance and size of webpack bundle
- delete unused code

There's not much to this PR, moved some things, deleted some unused code. We still bundle some code with webpack (driver, reporter). Ideally, we should not bundle those, but use Vite for everything. I'll try this next.

If I recall correctly, there are some circular deps in `driver` that webpack can handle (webpack goes off spec for ES modules for weird edge cases?), but Vite cannot. Another battle for another day.

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/app/cypress/e2e/runner/logger.cy.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import sinon from 'sinon'
```
Comment (lmiller1990) [inline packages/app/cypress/e2e/runner/logger.cy.ts:1]: Entire file is not new - just moved from `packages/runner-shared/src/logger.js` and turned into ES6 + TS.

File: packages/app/src/runner/logger.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+import trim from 'lodash/trim'
+import isString from 'lodash/isString'
+import isArray from 'lodash/isArray'
+import reduce from 'lodash/reduce'
+import result from 'lodash/result'
+import map from 'lodash/map'
+import sortBy from 'lodash/sortBy'
+import keys from 'lodash/keys'
+import isFunction from 'lodash/isFunction'
+
```
Comment (lmiller1990) [inline packages/app/src/runner/logger.ts:3]: Moved from `packages/runner-shared/src/logger.js` and turned into ES6 + TS.

File: packages/app/cypress/component/support/ctSupport.ts [hunk 3]
```diff
@@ -44,12 +41,9 @@ export const createTestAutIframe = (eventManager = createEventManager()) => {
     eventManager,
     _,
```
[the remark this answers is not in the document]
Comment (lmiller1990) [reply packages/app/cypress/component/support/ctSupport.ts:42]: Lodash import is still here, line 1.

File: packages/app/index.d.ts [hunk 4]
```diff
@@ -52,10 +41,7 @@ declare global {
        */
       React: any
       ReactDOM: any
-      logger: any
       dom: any
-      visitFailure: any
-      blankContents: any
       _: any // lodash
```
[the remark this answers is not in the document]
Comment (lmiller1990) [reply packages/app/index.d.ts:45]: Confusingly we have two copies of `_`. We can probably avoid this. I'll remove from the unified bundle.

File: packages/app/cypress/component/support/ctSupport.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (BlueWinds) [reply packages/app/cypress/component/support/ctSupport.ts:42]: Ah, so it is.

File: packages/app/index.d.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (lmiller1990) [reply packages/app/index.d.ts]: Done! https://github.com/cypress-io/cypress/pull/20069/commits/375a2fbea9cd847c2420352f9135d15b2ffc7859

**Comment by emilyrohrbough** [comment]
Do you have the  sizes of the webpack bundle before & after these changes?

**Comment by lmiller1990** [comment]
[See this comment for sizes](https://cypress-io.atlassian.net/browse/UNIFY-1047?focusedCommentId=15924) of development bundles. Here are the numbers for prod builds. This PR does not reduce (much) but it's a start - less code in the bundle makes it easier to debug and eventually eliminate. Numbers are in kB.

```sh
# lmiller1990/UNIFY-1047
$ du dist/cypress_runner.js         
8760	dist/cypress_runner.js

# 10.0-release
$ du dist/cypress_runner.js                              
8820	dist/cypress_runner.js
```

Both gzip and total bundle served is around 2MB.

Here is a breakdown of the bundle size. Lodash is huge, but we ship that as part of the public API, so we cannot remove it.

```
[
  {
    size: 544098,
    identifier: '/home/lachlan/code/work/cypress-10.0-release/node_modules/lodash/lodash.js',
    name: '/home/lachlan/code/work/cypress-10.0-release/node_modules/lodash/lodash.js'
  },
  {
    size: 538123,
    identifier: '/home/lachlan/code/work/cypress-10.0-release/node_modules/babel-loader/lib/index.js??ref--4!/home/lachlan/code/work/cypress-10.0-release/packages/runner-ct/src/index.js da3fe8c32f57e0cc4d9e4804182b1f4a',
    name: './src/index.js + 241 modules'
  },
  {
    size: 267194,
    identifier: '/home/lachlan/code/work/cypress-10.0-release/node_modules/jquery/dist/jquery.js',
    name: '/home/lachlan/code/work/cypress-10.0-release/node_modules/jquery/dist/jquery.js'
  },
  {
    size: 108169,
    identifier: '/home/lachlan/code/work/cypress-10.0-release/node_modules/react-dom/cjs/react-dom.production.min.js',
    name: '/home/lachlan/code/work/cypress-10.0-release/node_modules/react-dom/cjs/react-dom.production.min.js'
  },
  {
    size: 68493,
    identifier: '/home/lachlan/code/work/cypress-10.0-release/node_modules/mobx/lib/mobx.min.js',
    name: '/home/lachlan/code/work/cypress-10.0-release/node_modules/mobx/lib/mobx.min.js'
  },
  {
    size: 37965,
    identifier: '/home/lachlan/code/work/cypress-10.0-release/node_modules/babel-loader/lib/index.js??ref--4!/home/lachlan/code/work/cypress-10.0-release/packages/driver/src/dom/index.ts 1d89a08ac23a54ede3e29f85da5ca273',
    name: '../driver/src/dom/index.ts + 3 modules'
  },
  {
    size: 20195,
    identifier: '/home/lachlan/code/work/cypress-10.0-release/node_modules/css-element-queries/src/ElementQueries.js',
    name: '/home/lachlan/code/work/cypress-10.0-release/node_modules/css-element-queries/src/ElementQueries.js'
  },
  {
    size: 14890,
    identifier: '/home/lachlan/code/work/cypress-10.0-release/node_modules/events/events.js',
    name: '/home/lachlan/code/work/cypress-10.0-release/node_modules/events/events.js'
  },
  {
    size: 14668,
    identifier: '/home/lachlan/code/work/cypress-10.0-release/node_modules/ansi-to-html/lib/ansi_to_html.js',
    name: '/home/lachlan/code/work/cypress-10.0-release/node_modules/ansi-to-html/lib/ansi_to_html.js'
  },
  {
    size: 12235,
    identifier: '/home/lachlan/code/work/cypress-10.0-release/node_modules/css-element-queries/src/ResizeSensor.js',
    name: '/home/lachlan/code/work/cypress-10.0-release/node_modules/css-element-queries/src/ResizeSensor.js'
  },
  {
    size: 10855,
    identifier: '/home/lachlan/code/work/cypress-10.0-release/node_modules/mobx-react/dist/mobxreact.cjs.production.min.js',
    name: '/home/lachlan/code/work/cypress-10.0-release/node_modules/mobx-react/dist/mobxreact.cjs.production.min.js'
  },
  {
    size: 6831,
    identifier: '/home/lachlan/code/work/cypress-10.0-release/node_modules/react/cjs/react.production.min.js',
    name: '/home/lachlan/code/work/cypress-10.0-release/node_modules/react/cjs/react.production.min.js'
  },
  {
    size: 6813,
    identifier: '/home/lachlan/code/work/cypress-10.0-release/node_modules/@cypress/react-tooltip/dist/tooltip.js',
    name: '/home/lachlan/code/work/cypress-10.0-release/node_modules/@cypress/react-tooltip/dist/tooltip.js'
  },
  {
    size: 5418,
    identifier: '/home/lachlan/code/work/cypress-10.0-release/node_modules/process/browser.js',
    name: '/home/lachlan/code/work/cypress-10.0-release/node_modules/process/browser.js'
  },
  {
    size: 1363,
    identifier: '/home/lachlan/code/work/cypress-10.0-release/node_modules/react-dom/index.js',
    name: '/home/lachlan/code/work/cypress-10.0-release/node_modules/react-dom/index.js'
  },
  {
    size: 1328,
    identifier: '/home/lachlan/code/work/cypress-10.0-release/node_modules/classnames/index.js',
    name: '/home/lachlan/code/work/cypress-10.0-release/node_modules/classnames/index.js'
  },
  {
    size: 196,
    identifier: '/home/lachlan/code/work/cypress-10.0-release/node_modules/mobx-react/dist/index.js',
    name: '/home/lachlan/code/work/cypress-10.0-release/node_modules/mobx-react/dist/index.js'
  },
  {
    size: 190,
    identifier: '/home/lachlan/code/work/cypress-10.0-release/node_modules/react/index.js',
    name: '/home/lachlan/code/work/cypress-10.0-release/node_modules/react/index.js'
  },
  {
    size: 189,
    identifier: '/home/lachlan/code/work/cypress-10.0-release/node_modules/mobx/lib/index.js',
    name: '/home/lachlan/code/work/cypress-10.0-release/node_modules/mobx/lib/index.js'
  },
  {
    size: 28,
    identifier: 'multi /home/lachlan/code/work/cypress-10.0-release/packages/runner-ct/src/index.js',
    name: 'multi ./src/index.js'
  }
]
```

**Comment by lmiller1990** [comment]
@emilyrohrbough @BlueWinds can I get those 👀  and ✅ unless you see any other changes? Thanks!

**Review by BlueWinds — APPROVED** [review]
Ah, yeah. I'd approved before and just not come back to check it again.

**Review by elevatebart — APPROVED** [review]
